### PR TITLE
Check if input_bau.gdx is given or really needed; a couple of minor changes mostly related to MAgPIE coupling

### PR DESCRIPTION
**Pull request by dklein-pik**
## Purpose of this PR

This PR contains a couple of minor changes to R scripts mostly related to REMIND-MAgPIE coupling (code cleaning, adding comments, ...).

The most important change and not related to coupling is a new check in `checkFixCfg.R` that throws an error if an NDC is to be started but no `input_bau.gdx` was given. In case no NDC is to be calculated but an `input_bau.gdx` is provided it will be set to NA since it would otherwise introduce an unnecessary dependency to the preceeding BAU run. This is most relevant for the parallel mode of coupled runs, where in each iteration the unnecessary dependency may prevent the start of subseqent runs if the preceeding BAU run (that is actually not required in a non-NDC scenario) fails.

## Type of change

- [x] New feature 

## Checklist:

- [ ] My code follows the [coding etiquette](https://github.com/remindmodel/remind/blob/develop/main.gms#L80)
- [ ] I performed a self-review of my own code
- [ ] I explained my changes within the PR, particularly in hard-to-understand areas
- [ ] I checked that the [in-code documentation](https://github.com/remindmodel/remind/blob/develop/main.gms#L120) is up-to-date
- [ ] I adjusted the reporting in [`remind2`](https://github.com/pik-piam/remind2) where it was needed
- [ ] I adjusted `forbiddenColumnNames` in [readCheckScenarioConfig.R](https://github.com/remindmodel/remind/blob/develop/scripts/start/readCheckScenarioConfig.R) in case the PR leads to deprecated switches
- [x] All automated model tests pass (`FAIL 0` in the output of `make test`)
- [ ] The changelog `CHANGELOG.md` [has been updated correctly](https://gitlab.pik-potsdam.de/rse/rsewiki/-/wikis/Standards-for-Writing-a-Changelog)

### CODE REVIEW

File: scripts/start/checkFixCfg.R [hunk 1]
```diff
@@ -68,10 +73,20 @@ checkFixCfg <- function(cfg, remindPath = ".", testmode = FALSE) {
...
-  if (errorsfound > 0) {
-    if (testmode) warning(errorsfound, " errors found.")
-      else stop(errorsfound, " errors found, see above. Either adapt the parameter choice or the regexp in main.gms")
-  }
-
+  
+  # Make sure that an input_bau.gdx has been specified if an NDC is to be calculated.
+  if (isTRUE(cfg$gms$carbonprice == "NDC") | isTRUE(cfg$gms$carbonpriceRegi == "NDC")) {
+    if (is.na(cfg$files2export$start["input_bau.gdx"])) {
+      errormsg <- "'carbonprice' or 'carbonpriceRgi' is set to 'NDC' which requires a reference gdx in 'path_gdx_bau' but it is empty."
```
Comment (orichters) [inline scripts/start/checkFixCfg.R:80]: minor typo: carbonpriceR**e**gi

File: scripts/start/checkFixCfg.R [hunk 2]
```diff
@@ -68,10 +73,20 @@ checkFixCfg <- function(cfg, remindPath = ".", testmode = FALSE) {
...
+      if (testmode) warning(errormsg) else stop(errormsg)
+    }
+  } else {
+    if (!is.na(cfg$files2export$start["input_bau.gdx"])) {
+      message("Neither 'carbonprice' nor 'carbonpriceRgi' is set to 'NDC' but 'path_gdx_bau' ",
+              "is not empty introduing an unnecesary dependency to another run. Seeting 'path_gdx_bau' to NA")
+      cfg$files2export$start["input_bau.gdx"] <- NA
```
Comment (orichters) [inline scripts/start/checkFixCfg.R:87]: carbonpriceR**e**gi and introdu**c**ing lack some letters, and Se**t**ting has a typo.

File: tests/testthat/test_01-checkFixCfg.R [hunk 3]
```diff
@@ -7,8 +7,7 @@
 test_that("checkFixCfg works", {
   remind_folder <- "../.."
   savecfg <- cfg <- gms::readDefaultConfig(remind_folder)
-  # regexp = NA means: no warning
-  expect_warning(checkFixCfg(cfg, remind_folder, testmode = TRUE), regexp = NA)
+  expect_no_warning(checkFixCfg(cfg, remind_folder, testmode = TRUE))
```
Comment (orichters) [inline tests/testthat/test_01-checkFixCfg.R:10]: Wait, what? Did not know that existed, I hated the `regexp = NA` stuff. Turns out it was [just introduced](https://cran.r-project.org/web/packages/testthat/news/news.html) in version 3.1.5 as "experimental"… But if it works here, that is fine.

But you could think about adding `testthat (>= 3.1.5)` as dependency in `DESCRIPTION`, just to be safe.

**Review by orichters — APPROVED** [review]

File: scripts/start/checkFixCfg.R (hunk 2 above)
Comment (dklein-pik) [reply scripts/start/checkFixCfg.R:87]: And that was not even all. So "unnecesary"! OMG. Thanks!

**Pull request merged**